tracing-attributes/tests/tracing_override.rs

```diff
@@ -0,0 +1,76 @@
+//! Ensures that all unqualified uses of `tracing` can be overridden.
+//!
+//! As compilation fails directly when `tracing` is shadowed, no test functions are run here.
+
+use ::tracing::instrument;
+
+/// Shadows the crate `tracing` locally.
+#[allow(dead_code)]
+mod tracing {}
+
+#[instrument(tracing = ::tracing)]
+pub fn plain() {}
+
+#[instrument(tracing = ::tracing)]
+pub async fn async_fn() {}
+
+#[derive(Debug)]
+pub struct Custom;
+
+#[instrument(tracing = ::tracing)]
+pub fn implicit_field_debug(custom: Custom) {}
+
+#[instrument(tracing = ::tracing, level = "error")]
+pub fn level_error() {}
+#[instrument(tracing = ::tracing, level = "warn")]
+pub fn level_warn() {}
+#[instrument(tracing = ::tracing, level = "info")]
+pub fn level_info() {}
+#[instrument(tracing = ::tracing, level = "debug")]
+pub fn level_debug() {}
+#[instrument(tracing = ::tracing, level = "trace")]
+pub fn level_trace() {}
+
+#[instrument(tracing = ::tracing, level = 5)]
+pub fn level_5() {}
+#[instrument(tracing = ::tracing, level = 4)]
+pub fn level_4() {}
+#[instrument(tracing = ::tracing, level = 3)]
+pub fn level_3() {}
+#[instrument(tracing = ::tracing, level = 2)]
+pub fn level_2() {}
+#[instrument(tracing = ::tracing, level = 1)]
+pub fn level_1() {}
+
+const A_LEVEL: ::tracing::Level = ::tracing::Level::INFO;
+#[instrument(tracing = ::tracing, level = A_LEVEL)]
+pub fn level_ident() {}
```

Comment:
The ability to specify the level through an integer constant or `Level`-identifier seem to be undocumented, as far as the immediate `///`-documentation on `#[instrument]` goes. Should I document it or is this deprecated/out of scope for this PR?

(I'd still like to keep the test here regardless of the answer to that question, since they all have distinct code paths.)

Reply:
It should probably be documented, but in a follow-up PR. We should probably document it on the `Level` type as well, since I believe its `FromStr` implementation will also accept numbers.